app/src/main/kotlin/com/wire/android/ui/home/HomeScreen.kt

```diff
@@ -103,20 +155,53 @@ class HomeState(
         homeBottomSheetContent = content
     }
 
+    fun updateScrollPosition(newScrollPosition: Int) {
+        scrollPosition = newScrollPosition
+    }
+
+    fun closeDrawer() {
+        coroutineScope.launch {
+            drawerState.close()
+        }
+    }
+
+    fun openDrawer() {
+        coroutineScope.launch {
+            drawerState.open()
+        }
+    }
+
 }
 
-@OptIn(ExperimentalMaterialApi::class)
+@OptIn(ExperimentalMaterialApi::class, ExperimentalMaterial3Api::class)
 @Composable
 fun rememberHomeState(
     coroutineScope: CoroutineScope = rememberCoroutineScope(),
+    navController: NavHostController = rememberNavController(),
+    drawerState: DrawerState = rememberDrawerState(DrawerValue.Closed),
     bottomSheetState: ModalBottomSheetState = rememberModalBottomSheetState(initialValue = ModalBottomSheetValue.Hidden),
     bottomSheetContent: @Composable (ColumnScope.() -> Unit)? = null
 ): HomeState {
-    return remember {
+    val navBackStackEntry by navController.currentBackStackEntryAsState()
+
+    val navigationItem = when (navBackStackEntry?.destination?.route) {
+        HomeNavigationItem.Archive.route -> HomeNavigationItem.Archive
+        HomeNavigationItem.Vault.route -> HomeNavigationItem.Vault
+        else -> HomeNavigationItem.Conversations
+    }
```

Comment:
> Can we just use `HomeNavigationItem.values().first { it.route == navBackStackEntry?.destination?.route }`? 🤔 
> or at least implement all branches directly instead of using `else` -  this way we won't forget about this when for instance we add another item to this enum

https://github.com/wireapp/wire-android-reloaded/pull/410/files#r819372276